feat(self-hosted): Add docs around backup/restore (#258)
getsentry/develop: src/docs/self-hosted/backup.mdx

---
title: "Self-Hosted Backup & Restore"
---

> ## Backup via export

Reviewer: ```suggestion
## Backup With `export`
```

If you need a quick way to backup and restore your Sentry instance and you don't need historical event data, you can use the built in `export` and `import` commands. These commands will save and load all project and user data, but will not contain any event data.

```shell
docker-compose run --rm -T -e SENTRY_LOG_LEVEL=CRITICAL web export > backup.json
```

If you omit the `-T` or `-e SENTRY_LOG_LEVEL=CRITICAL` parts, your backup file will have logging lines mixed in and you'll have to remove them somehow.

> ## Restore via import

Reviewer: ```suggestion
## Restore With `import`
```

> Once you have a backup via the `export` command, the easiest way to restore it is to placing it under the `sentry` directory in your main `onpremise` repo, next to the config files. This directory automatically gets mounted to `/etc/sentry` so you can run the following to restore your data:

Reviewer: ```suggestion
Once you have a backup by using the `export` command, the easiest way to restore it is to place it under the `sentry` directory in your main `onpremise` repo, next to the config files. This directory automatically gets mounted to `/etc/sentry` so you can run the following to restore your data:
```

```shell
docker-compose run --rm -T web import /etc/sentry/backup.json
```

If you don't see any errors and the process exits with code `0`, congratulations, you have just restored your data.

> It is strongly recommended to restore the data on the same version of Sentry on a fresh install (empty database but migrations are run), as otherwise you are very likely to hit errors and even corrupt your database.

Reviewer: ```suggestion
We strongly recommended that you restore data on the **same version of Sentry** on a fresh install (empty database but migrations are run). Otherwise, you are very likely to hit errors and may corrupt your database.
```

> ## Total backup/restore via Docker volumes

Reviewer: ```suggestion
## Total Backup and Restore With Docker Volumes
```

> The ideal way to backup and restore Sentry is to backup and restore all Docker volumes used by it. All volumes that hold critical long-term data are already defined as global volumes at install time and are prefixed with `sentry-`:

Reviewer: ```suggestion
The ideal way to backup and restore Sentry is to backup and restore all Docker volumes that it uses. All volumes that hold critical long-term data are already defined as global volumes at install time and are prefixed with `sentry-`:
```
